subsys/bluetooth/shell/bt.c

```diff
@@ -4104,7 +4104,7 @@ SHELL_STATIC_SUBCMD_SET_CREATE(bt_cmds,
 	SHELL_CMD_ARG(phy-update, NULL, "<tx_phy> [rx_phy] [s2] [s8]",
 		      cmd_conn_phy_update, 2, 3),
 #endif
-#if defined(CONFIG_BT_CENTRAL)
+#if defined(CONFIG_BT_CENTRAL) || defined(CONFIG_BT_BROADCASTER)
 	SHELL_CMD_ARG(channel-map, NULL, "<channel-map: XXXXXXXXXX> (36-0)",
 		      cmd_chan_map, 2, 1),
 #endif /* CONFIG_BT_CENTRAL */
```

Comment:
Since the function itself is not guarded (but will just return an error), do we need the guard here at all?

Reply:
Is it not guarded by `CONFIG_BT_CENTRAL`? https://github.com/zephyrproject-rtos/zephyr/blob/3d29c9fe546cd42554becee9dddcbe2e9e73d5cc/subsys/bluetooth/shell/bt.c#L3236-L3237

Reply:
@adamfc2000
`cmd_chan_map` is not built if only CONFIG_BT_BROADCASTER is enabled (and CONFIG_BT_CENTRAL is not). Could this be fixed?